### PR TITLE
Provide a single option to disable all network access and testing.

### PR DESCRIPTION
**Pull request by DennisHeimbigner**
Add the option "--disable-network-access" (automake) or "-DENABLE_NETWORK_ACCESS=OFF" (cmake).
When disabled, this option transitively disables all network access capabilities and testing.
If set, this option implies the following:
* --disable-dap
* --disable-byterange
* --disable-s3

This PR answers a request for a feature from Ed Hartnett.

## Misc. Other changes
* Take the opportunity to clean up some old, unused options; e.g. --enable-multifilters.
* Fix bug in using S3 urls.

### CODE REVIEW

File: include/netcdf_meta.h.in
Comment (WardF) [inline include/netcdf_meta.h.in:1]: Won't removing this define potentially cause problems for anybody who is currently using it?

**Review by WardF — COMMENTED** [review]
I'm concerned that 'network access' might be ambiguous; I can imagine questions, down the line, around 'what about networked drives? What about X?'.  I think a single-switch shortcut which acts as a master switch to set existing remote data access options to off is a fine idea, but I wonder about overstating what it's doing.

File: libnetcdf.settings.in [hunk 1]
```diff
@@ -51,11 +51,10 @@ JNA Support:		@HAS_JNA@
...
 
-Multi-Filter Support:	@HAS_MULTIFILTERS@
 Quantization:		@HAS_QUANTIZE@
 Logging:     		@HAS_LOGGING@
 SZIP Write Support:     @HAS_SZLIB_WRITE@
 Standard Filters:       @STD_FILTERS@
 ZSTD Support:           @HAS_ZSTD@
 Parallel Filters:       @HAS_PAR_FILTERS@
-
+Allow Network Access:   @DO_NETWORK_ACCESS@
```
Comment (WardF) [inline libnetcdf.settings.in:60]: I will make a change to this summary, I don't want to mislead or overstate what we're doing; 'Network Access' is ambiguous and can mean a *lot*.

**Review by WardF — APPROVED** [review]

**Review by DennisHeimbigner — COMMENTED** [review]
This seems like a good choice.

**Pull request merged**